graf3d/CMakeLists.txt

```diff
@@ -30,5 +30,7 @@ if (opengl)
 endif()
 
 if(webgui AND root7)
-  add_subdirectory(eve7) # special CMakeLists.txt
+  if(NOT (MSVC AND CMAKE_GENERATOR MATCHES Ninja AND CMAKE_BUILD_TYPE MATCHES Debug))
```

Comment:
We should add a comment explaining why `eve7` does not work with `MSVC +  Ninja + Debug`.

Reply:
Indeed, but I'm still working on it. And the I/O is still broken in debug mode, due to the special iterators on x64, remember? If that could be fixed, I could remove some limiting and annoying workarounds...

Reply:
And the error is:
```
[376/617] Generating G__ROOTEve.cxx, ../../bin/libROOTEve_rdict.pcm, ../../bin/libROOTEve.rootmap
FAILED: graf3d/eve7/G__ROOTEve.cxx bin/libROOTEve_rdict.pcm bin/libROOTEve.rootmap C:/root-dev/build/x64/ninja/graf3d/eve7/G__ROOTEve.cxx C:/root-dev/build/x64/ninja/bin/libROOTEve_rdict.pcm C:/root-dev/build/x64/ninja/bin/libROOTEve.rootmap
C:\windows\system32\cmd.exe /C "cd /D C:\root-dev\build\x64\ninja\graf3d\eve7 && "C:\Program Files\CMake\bin\cmake.exe" -E env LD_LIBRARY_PATH=C:/root-dev/build/x64/ninja/lib: ROOTIGNOREPREFIX=1 C:/root-dev/build/x64/ninja/bin/rootcling.exe -rootbuild -v2 -f G__ROOTEve.cxx -s C:/root-dev/build/x64/ninja/bin/libROOTEve.dll -m libCore_rdict.pcm -m libGeom_rdict.pcm -m libPhysics_rdict.pcm -m libEG_rdict.pcm -m libTreePlayer_rdict.pcm -m libRCsg_rdict.pcm -m libROOTWebDisplay_rdict.pcm -excludePath C:/root-dev/git/master -excludePath C:/root-dev/build/x64/ninja/ginclude -excludePath C:/root-dev/build/x64/ninja/externals -excludePath C:/root-dev/build/x64/ninja/builtins -rml libROOTEve.dll -rmf C:/root-dev/build/x64/ninja/bin/libROOTEve.rootmap -DNO_SSL -DHTTP_WITHOUT_FASTCGI -IC:/root-dev/build/x64/ninja/include -IC:/root-dev/build/x64/ninja/LZMA/src/LZMA/include -IC:/root-dev/git/master/graf3d/eve7/inc -IC:/root-dev/git/master/core/winnt/inc -IC:/root-dev/git/master/builtins/zstd -IC:/root-dev/git/master/core/zstd/inc -IC:/root-dev/git/master/core/lz4/inc -IC:/root-dev/build/x64/ninja/LZMA/src/LZMA/include -IC:/root-dev/git/master/core/lzma/inc -IC:/root-dev/git/master/core/zip/inc -IC:/root-dev/git/master/core/thread/inc -IC:/root-dev/git/master/core/textinput/src -IC:/root-dev/git/master/core/textinput/inc -IC:/root-dev/git/master/core/rint/inc -IC:/root-dev/git/master/core/meta/inc -IC:/root-dev/git/master/core/gui/inc -IC:/root-dev/git/master/core/foundation/res -IC:/root-dev/git/master/core/foundation/v7/inc -IC:/root-dev/git/master/core/foundation/inc -IC:/root-dev/git/master/core/cont/inc -IC:/root-dev/git/master/core/clingutils/res -IC:/root-dev/git/master/core/clingutils/inc -IC:/root-dev/git/master/core/clib/inc -IC:/root-dev/git/master/core/base/v7/inc -IC:/root-dev/git/master/core/base/inc -IC:/root-dev/build/x64/ninja/ginclude -IC:/root-dev/git/master/math/physics/inc -IC:/root-dev/git/master/math/matrix/inc -IC:/root-dev/git/master/math/mathcore/v7/inc -IC:/root-dev/git/master/math/mathcore/inc -IC:/root-dev/git/master/core/imt/inc -IC:/root-dev/git/master/tree/treeplayer/inc -IC:/root-dev/git/master/hist/hist/inc -IC:/root-dev/git/master/tree/tree/inc -IC:/root-dev/git/master/graf3d/csg/inc -IC:/root-dev/git/master/builtins -IC:/root-dev/git/master/graf3d/eve7 -IC:/root-dev/git/master/geom/geom/inc -IC:/root-dev/git/master/io/io/v7/inc -IC:/root-dev/git/master/io/io/inc -IC:/root-dev/git/master/math/genvector/inc -IC:/root-dev/git/master/montecarlo/eg/inc -IC:/root-dev/git/master/graf2d/gpad/inc -IC:/root-dev/git/master/graf2d/graf/inc -IC:/root-dev/git/master/graf3d/g3d/inc -IC:/root-dev/git/master/net/net/inc -IC:/root-dev/git/master/gui/webdisplay/inc -IC:/root-dev/git/master/net/http/inc ROOT/REveChunkManager.hxx ROOT/REveClient.hxx ROOT/REveCompound.hxx ROOT/REveBox.hxx ROOT/REveBoxSet.hxx ROOT/REveCaloData.hxx ROOT/REveCalo.hxx ROOT/REveDataCollection.hxx ROOT/REveDataProxyBuilderBase.hxx ROOT/REveDataSimpleProxyBuilder.hxx ROOT/REveDataSimpleProxyBuilderTemplate.hxx ROOT/REveDataTable.hxx ROOT/REveDigitSet.hxx ROOT/REveElement.hxx ROOT/REveFrameBox.hxx ROOT/REveGeoPolyShape.hxx ROOT/REveGeoShapeExtract.hxx ROOT/REveGeoShape.hxx ROOT/REveGluTess.hxx ROOT/REveJetCone.hxx ROOT/REveEllipsoid.hxx ROOT/REveLine.hxx ROOT/REveManager.hxx ROOT/REvePathMark.hxx ROOT/REvePointSet.hxx ROOT/REvePolygonSetProjected.hxx ROOT/REveProjectionBases.hxx ROOT/REveProjectionManager.hxx ROOT/REveProjections.hxx ROOT/REveRenderData.hxx ROOT/REveRGBAPalette.hxx ROOT/REveScalableStraightLineSet.hxx ROOT/REveScene.hxx ROOT/REveSceneInfo.hxx ROOT/REveSecondarySelectable.hxx ROOT/REveSelection.hxx ROOT/REveShape.hxx ROOT/REveStraightLineSet.hxx ROOT/REveSystem.hxx ROOT/REveTableInfo.hxx ROOT/REveTableProxyBuilder.hxx ROOT/REveTrack.hxx ROOT/REveTrackProjected.hxx ROOT/REveTrackPropagator.hxx ROOT/REveTrans.hxx ROOT/REveTreeTools.hxx ROOT/REveTypes.hxx ROOT/REveUtil.hxx ROOT/REveVector.hxx ROOT/REveViewer.hxx ROOT/REveViewContext.hxx ROOT/REveVSD.hxx ROOT/REveVSDStructs.hxx C:/root-dev/git/master/graf3d/eve7/inc/LinkDef.h"
Access violation
```

Reply:
That explains the part `if(NOT (MSVC AND CMAKE_BUILD_TYPE MATCHES Debug))` but not the part `AND CMAKE_GENERATOR MATCHES Ninja`.  Either way we should add a comment.

Reply:
@pcanal  You're commenting on deleted code...